Gems/AWSCore/Code/Source/Editor/UI/AWSCoreResourceMappingToolAction.cpp

```diff
@@ -11,124 +11,141 @@
+                char* lastSeparatorAddress = strrchr(executablePath, AZ_CORRECT_FILESYSTEM_SEPARATOR);
+                if (lastSeparatorAddress)
                 {
-                    // Remove the file name if it exists, and keep the parent folder only
-                    char* lastSeparatorAddress = strrchr(executablePath, AZ_CORRECT_FILESYSTEM_SEPARATOR);
-                    if (lastSeparatorAddress)
-                    {
-                        *lastSeparatorAddress = '\0';
-                    }
+                    *lastSeparatorAddress = '\0';
                 }
             }
+        }
 
-            AZStd::string binDirectoryPath(executablePath);
-            auto lastSeparator = binDirectoryPath.find_last_of(AZ_CORRECT_FILESYSTEM_SEPARATOR);
-            if (lastSeparator != AZStd::string::npos)
-            {
-                m_isDebug = binDirectoryPath.substr(lastSeparator).contains("debug");
-            }
+        AZStd::string binDirectoryPath(executablePath);
+        auto lastSeparator = binDirectoryPath.find_last_of(AZ_CORRECT_FILESYSTEM_SEPARATOR);
+        if (lastSeparator != AZStd::string::npos)
+        {
+            m_isDebug = binDirectoryPath.substr(lastSeparator).contains("debug");
+        }
 
-            m_toolQtBinDirectoryPath = AZStd::string::format("%s/%s", binDirectoryPath.c_str(), "AWSCoreEditorQtBin");
-            AzFramework::StringFunc::Path::Normalize(m_toolQtBinDirectoryPath);
-            if (!AZ::IO::SystemFile::Exists(m_toolQtBinDirectoryPath.c_str()))
-            {
-                AZ_Error("AWSCoreEditor", false, "Failed to find ResourceMappingTool Qt binaries at %s.", m_toolQtBinDirectoryPath.c_str());
-                m_toolQtBinDirectoryPath.clear();
-            }
+        m_toolQtBinDirectoryPath = AZStd::string::format("%s/%s", binDirectoryPath.c_str(), "AWSCoreEditorQtBin");
+        AzFramework::StringFunc::Path::Normalize(m_toolQtBinDirectoryPath);
+        if (!AZ::IO::SystemFile::Exists(m_toolQtBinDirectoryPath.c_str()))
+        {
+            AZ_Error(
+                AWSCoreResourceMappingToolActionName, false, "Failed to find ResourceMappingTool Qt binaries at %s.",
+                m_toolQtBinDirectoryPath.c_str());
+            m_toolQtBinDirectoryPath.clear();
         }
     }
 
     AZStd::string AWSCoreResourceMappingToolAction::GetToolLaunchCommand() const
     {
-        if (m_enginePythonEntryPath.empty() || m_toolScriptPath.empty() || m_toolQtBinDirectoryPath.empty())
+        if (m_enginePythonEntryPath.empty() ||
+            m_toolScriptPath.empty() ||
+            m_toolQtBinDirectoryPath.empty() ||
+            m_toolConfigDirectoryPath.empty() ||
+            m_toolLogDirectoryPath.empty())
         {
+            AZ_Error(AWSCoreResourceMappingToolActionName, false,
+                "Failed to get tool launch command, engine python path: %s, tool script path: %s, tool qt binaries path: %s, tool config path: %s, tool log path: %s",
```

Comment:
minor: "Expected parameter for tool launch command is missing. Engine python ..."